release notes/v0.30.0.md

```diff
@@ -0,0 +1,123 @@
+    let user = users[randomUserID]; // alternativaly, we can also use __VU and/or __ITER
+    console.log(`VU ${__VU} is running iteration ${__ITER} with user ${user.username}...`);
+    http.post('https://httpbin.test.k6.io/post', JSON.stringify(user));
+    sleep(Math.random() * 2); // or, better yet, use arribal-rate
+}
+```
+
+Notice how `Loading users.json` is logged only once, but each VU uses the `users` variable like a normal JS array. The data is read only once and we have just a single copy of the huge array in memory! Behind the scenes, k6 uses a JS [`Proxy`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy) to transparently copy only the row each VU requests in `users[randomUserID]` to it. This on-demand copying is a bit inefficient, but it's still leagues better than having a copy of the huge array in every VU! And you can avoid the copying in every iteration by pinning the data used by every VU and having `let user = users[randomUserID]` in the [init context](https://k6.io/docs/using-k6/test-life-cycle)!
+
+And yes, you can have multiple `SharedArray` objects in the same script, just make sure to give them unique names - this is what `someName` in the script above was for. Because VUs are independent JS runtimes, we need some way to differentiate between the different shared memory objects, so we require them to have unique names. These names are also the IDs that any [xk6](https://github.com/k6io/xk6) extensions would need to use to access them.
+
+This works both locally and in the cloud. We advise everyone that deals with huge data files to wrap them in a `SharedArray` and give this new feature a try. The required script changes should be minimal, while the benefits should be significant. Hopefully, we can finally consider one of the biggest blockers k6 users have had for a long time solved! :tada:
+
+### Support a `handleSummary()` callback at the end of the test ([#1768](https://github.com/loadimpact/k6/pull/1768))
+
+You can now `export` a function called `handleSummary()` and k6 will call it at the end of the test run, after even `teardown()`. `handleSummary()` will be called with a JS object containing the same information that is used to generate the end-of-test summary and `--summary-export`, and allows users to completely customize how the end-of-test summary looks like.
+
+Besides customizing the end-of-test CLI summary (if `handleSummary()` is exported, k6 will not print the default), you can also generate and save as files various machine-readable data formats with the same data (JSON, CVS, JUnit/xUnit/etc. XML, etc.), as well as human-readable formats like HTML. Even binary formats like PDF are not out of reach, potentially, with an appropriate JS library that works in k6! You can also send the generated reports to a remote server by making an HTTP request with them (or using any of the other protocols k6 already supports)!
```

Comment:
```suggestion
Besides customizing the end-of-test CLI summary (if `handleSummary()` is exported, k6 will not print the default), you can also generate and save to files various machine-readable data formats with the same data (JSON, CVS, JUnit/xUnit/etc. XML, etc.), as well as human-readable formats like HTML. Even binary formats like PDF are not out of reach, potentially, with an appropriate JS library that works in k6! You can also send the generated reports to a remote server by making an HTTP request with them (or using any of the other protocols k6 already supports)!
```
This just seems more natural to me ... 


But also the whole thing reads like there is built-in way to export to all of those. I think it should be reworded so that confusion is not possible. 

I think something like: "you can also generate and save data to files in both text and binary form. This does allow the creation of js helpers to generate <formats> that aren't provided by k6 natively but here is a list of community-made ones." (and the last one is obviously a link to the documentation with just the most basic of junit one that I've hacked together).

Reply:
resolved by https://github.com/loadimpact/k6/pull/1809/commits/78875f72a82841ecb179455f9c891c87526a5424?